cmd/system/aws_cli.go

```diff
@@ -0,0 +1,317 @@
+			{{$ext = "msi"}}
+			{{ else if eq .OS "Darwin" -}}
+			{{$uri = "AWSCLIV2"}}
+			{{$ext = "pkg"}}
+			{{ end -}}
+
+			{{$base}}/{{$uri}}-{{$version}}.{{$ext}}
+		`,
+		}
+
+		dUrl, err := awsCliTool.GetURL(osVer, arch, version, !progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloading from: %s\n", dUrl)
+
+		outPath, err := get.DownloadFileP(dUrl, progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		if workDir == "" {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			workDir = cwd
+		}
+
+		fmt.Printf("Copying file to: %s\n", workDir)
+
+		filename := filepath.Base(outPath)
+		if _, err = get.CopyFileP(
+			outPath,
+			fmt.Sprintf("%s/%s", workDir, filename), readWriteExecuteEveryone,
+		); err != nil {
+			return err
+		}
+
+		extension := filepath.Ext(outPath)
+		if extension == ".zip" {
+			unpackPath := fmt.Sprintf("%s/awscli", workDir)
+			fmt.Printf("Unpacking AWS CLI to: %s\n", unpackPath)
+
+			fInfo, err := f.Stat()
+			if err != nil {
+				return err
+			}
+			if err := archive.Unzip(f, fInfo.Size(), unpackPath, true); err != nil {
+				return err
+			}
+
+			workDir = unpackPath
+		}
+
+		if runInstaller {
+			if err := runBundledInstaller(osVer, workDir, filename, installPath); err != nil {
+				return err
+			}
+		} else {
+			tpl, err := installationInstructions(osVer, workDir, filename, installPath)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("\n%s", tpl)
+		}
+
+		return nil
+	}
+
+	return command
+}
+
+func getAWSCLIVersion(owner, repo string) (string, error) {
+	url := fmt.Sprintf("https://github.com/%s/%s", owner, repo)
+
+	client := http.Client{Timeout: time.Second * 10}
+	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+
+	remote := git.NewRemote(memory.NewStorage(), &config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{url},
+	})
+
+	references, err := remote.List(&git.ListOptions{PeelingOption: git.AppendPeeled})
+	if err != nil {
+		return "", err
+	}
+
+	tags := make([]*semver.Version, 0)
+	for _, reference := range references {
+		if reference.Name().IsTag() {
+			trimmed := strings.TrimPrefix(reference.Name().String(), "refs/tags/")
+			v, err := semver.NewVersion(trimmed)
+			if err != nil && errors.Is(err, semver.ErrInvalidSemVer) {
+				continue
+			}
+
+			tags = append(tags, v)
+		}
+	}
+
+	var comparer = func(a, b *semver.Version) int {
+		return a.Compare(b)
+	}
+
+	slices.SortFunc(tags, comparer)
+	latest := tags[len(tags)-1]
+
+	return latest.String(), nil
+}
+
+func runBundledInstaller(osVer string, workDir string, filename string, installPath string) error {
+	fmt.Printf("Running bundled installer from download URL\n")
+
+	cmd := ""
+	args := make([]string, 0)
+
+	switch osVer {
+	case "Darwin":
+		cmd = "sudo"
```

Comment:
I would much prefer to asking for an explicit `sudo` and check the privileges on the `PreRunE` section, then implicitly calling it in the command.

Reply:
I've addressed this by adding a prompt that sets a `runElevated` variable that is checked for `true` before running the bundled installer, otherwise falling back to printing the instructions, is this sufficient or does it need further re-thinking?

Reply:
Nice idea, but I would stick to run the `sudo arakde system install aws-cli` much rather then making a prompt.

Seems like the other commands require this too [eg: firecracker](https://github.com/alexellis/arkade/blob/master/docs/arkade.tape#L58)